pkg/policy/opa/engine.go

```diff
@@ -249,14 +249,20 @@ func (e *Engine) CompileRegoFiles() error {
 
 // Init initializes the Opa engine
 // Handles loading all rules, filtering, compiling, and preparing for evaluation
-func (e *Engine) Init(policyPath string) error {
+func (e *Engine) Init(policyPath string, scanRules, skipRules []string) error {
 	e.context = context.Background()
 
 	if err := e.LoadRegoFiles(policyPath); err != nil {
 		zap.S().Error("error loading rego files", zap.String("policy path", policyPath), zap.Error(err))
 		return err
 	}
 
+	// before compiling the rego files, filter the rules based on scan and skip rules supplied
+	filterRules(e, policyPath, scanRules, skipRules)
```

Comment:
Would it make sense to have `filterRules` to be tied to the `Engine` struct?

Reply:
implemented as part of interface

Reply:
At this point, adding to an interface makes sense. Later, we should have a top-level "Scanner" struct that manages interactions between the scan engine, filters, policies/rules, and results. Today, that's all considered the "scan engine", but I just want to limit that piece to actual scanning only. Policies, rules, rule filters, and results processing are applicable to different scan engines, so in the future they should work independently of it.